services/python/python-version-from-toml.service.js

```diff
@@ -0,0 +1,65 @@
+import Joi from 'joi'
+import BaseTomlService from '../../core/base-service/base-toml.js'
+import { optionalUrl } from '../validators.js'
+
+const queryParamSchema = Joi.object({
+  tomlFilePath: optionalUrl.required(),
+}).required()
+
+const schema = Joi.object({
+  project: Joi.object({
+    'requires-python': Joi.string().required(),
+  }).required(),
+}).required()
+
+const documentation = `Shows the required python version for a package based on the values in the requires-python field in the pyproject.toml \n
+a URL of the toml is required, please note that when linking to files in github or similar sites, provide URL to raw file, for example:
+
+Use https://raw.githubusercontent.com/numpy/numpy/main/pyproject.toml \n
+And not https://github.com/numpy/numpy/blob/main/pyproject.toml
+`
+
+class PythonVersionFromToml extends BaseTomlService {
+  static category = 'platform-support'
+
+  static route = {
+    base: '',
+    pattern: 'python/required-version-toml',
+    queryParamSchema,
+  }
+
+  static examples = [
+    {
+      title: 'Python Required Version TOML',
```

Comment:
Can we clarify in the title and description that this is for PEP 621-compliant `pyproject.toml`

Python packaging is..fun :grimacing:  and lots of projects have a pyproject.toml that doesn't use this format e.g: https://github.com/chris48s/arcgis2geojson/blob/master/pyproject.toml#L12

Reply:
Fixed with https://github.com/badges/shields/pull/9516/commits/e2ecbf5ff2a83f419307f752c453460be345b027